src/settings/LanguagePicker.js

```diff
@@ -15,23 +15,53 @@ const styles = StyleSheet.create({
 type Props = {|
   value: string,
   onValueChange: (locale: string) => void,
+  filter: string,
 |};
 
-export default class LanguagePicker extends PureComponent<Props> {
+type State = {|
+  data: Array<*>,
+|};
+
+export default class LanguagePicker extends PureComponent<Props, State> {
+  state = {
+    data: languages,
+  };
+
+  handleFilterChange = (text: string) => {
+    const newData = languages.filter(item => {
```

Comment:
Run this code in `render()` and you can remove `componentDidUpdate()`

Reply:
Updated!